src/overlays/actors/ovl_En_Bigpamet/z_en_bigpamet.c

```diff
@@ -90,121 +113,763 @@ static DamageTable D_80A29734 = {
+    Vec3f sp8C;
+    f32 temp_fs0;
+    f32 temp_fs4;
+    f32 temp_fs5;
+    s16 temp_s0;
+
+    temp_fs4 = Math_SinS(this->actor.wallYaw + 0x4000);
+    temp_fs5 = Math_CosS(this->actor.wallYaw + 0x4000);
+
+    sp8C.x = Math_SinS(this->actor.wallYaw + 0x8000) * 50.0f + this->actor.world.pos.x;
+    sp8C.y = this->actor.world.pos.y;
+    sp8C.z = (Math_CosS(this->actor.wallYaw + 0x8000) * 50.0f) + this->actor.world.pos.z;
+
+    sp8C.x += ((this->actor.world.pos.x - sp8C.x) * 0.3f);
+    sp8C.z += ((this->actor.world.pos.z - sp8C.z) * 0.3f);
+
+    for (i = 0; i < 4; i++) {
+        temp_fs0 = Rand_ZeroFloat(60.0f) + 50.0f;
+        temp_s0 = Rand_Next() >> 0x11;
+
+        sp98.x = (Math_CosS(temp_s0) * temp_fs0 * temp_fs4) + sp8C.x;
+        sp98.y = (Math_SinS(temp_s0) * temp_fs0) + sp8C.y;
+        sp98.z = (Math_CosS(temp_s0) * temp_fs0 * temp_fs5) + sp8C.z;
+
+        func_800B12F0(play, &sp98, &gZeroVec3f, &gZeroVec3f, Rand_S16Offset(950, 0x64), Rand_S16Offset(0x14, 0xA), 20);
+    }
+}
+
+void func_80A27FE8(EnBigpamet* this, PlayState* play) {
+    Vec3f sp34;
+    s16 sp32;
+
+    if (this->actor.depthInWater > 0.0f) {
+        sp34.x = this->actor.world.pos.x;
+        sp34.z = this->actor.world.pos.z;
+        sp34.y = this->actor.world.pos.y + this->actor.depthInWater;
+
+        EffectSsGRipple_Spawn(play, &sp34, 500, 900, 0);
+        sp34.y += 8.0f;
+
+        if (this->actionFunc != func_80A28D0C) {
+            sp32 = (s32)randPlusMinusPoint5Scaled(32768.0f) + this->actor.world.rot.y;
+            sp34.x -= 55.0f * Math_SinS(sp32);
+            sp34.z -= 55.0f * Math_CosS(sp32);
+        }
+        EffectSsGSplash_Spawn(play, &sp34, NULL, NULL, 0, Rand_S16Offset(1400, 0xC8));
+    }
+}
+
+void func_80A2811C(EnBigpamet* this, PlayState* play) {
+    Vec3f sp2C;
+
+    if ((this->actor.depthInWater > 0.0f) && ((play->gameplayFrames % 14) == 0)) {
+        sp2C.x = this->actor.world.pos.x;
+        sp2C.y = this->actor.world.pos.y + this->actor.depthInWater;
+        sp2C.z = this->actor.world.pos.z;
+
+        EffectSsGRipple_Spawn(play, &sp2C, 500, 900, 0);
+    }
+}
+
+void func_80A281B4(EnBigpamet* this) {
+    this->actor.draw = NULL;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->actionFunc = func_80A281DC;
+}
+
+void func_80A281DC(EnBigpamet* this, PlayState* play) {
+    WaterBox* sp2C;
+    f32 sp28;
+
+    if (this->actor.parent->params == 1) {
+        if (WaterBox_GetSurface1(play, &play->colCtx, this->actor.world.pos.x, this->actor.world.pos.z, &sp28, &sp2C)) {
+            this->actor.depthInWater = sp28 - this->actor.world.pos.y;
+        } else {
+            this->actor.depthInWater = this->actor.world.pos.y;
+        }
+
+        func_80A2778C(this);
+        func_80A27970(this, play);
+        func_80A28274(this);
+    }
+}
+
+void func_80A28274(EnBigpamet* this) {
+    Animation_PlayLoop(&this->skelAnime1, &object_tl_Anim_004210);
+    this->actor.draw = EnBigpamet_Draw;
+    this->unk_29E = 0x14;
```

Comment:
```suggestion
    this->unk_29E = 20;
```
Decremented next function, timer